src/components/TimeChart.js

```diff
@@ -70,34 +71,46 @@ class TimeChart extends React.Component {
         .x(d => x(d.date))
         .y(d => y(d.value))
 
+    let active
     let callout
     if (hover) {
       const bisectDate = bisector(d => d.date).left
       const x0 = x.invert(hover.x * width)
       const i = bisectDate(dataClean, x0, 1)
       const [d0, d1] = [dataClean[i - 1], dataClean[i]]
-      const dActive = x0 - d0.date > d1.date - x0 ? d1 : d0
-
-      callout = (
-        <g transform={`translate(${x(dActive.date)}, 0)`}>
-          <line y2={height} stroke='#000' strokeWidth='1' />
-          <text x='4' dy='.71em' className='h6 monospace'>
-            {keys.map(k => `${k}: ${dActive[k]}`).join(', ')}
-          </text>
-        </g>
-      )
+
+      if (d0 && d1) {
+        active = x0 - d0.date > d1.date - x0 ? d1 : d0
```

Comment:
This conditional is pretty hard to understand because of the operators. Maybe include `()` where appropriate so that the different parts are a bit easier to understand